src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -85,6 +143,15 @@ trait DataUseRestrictionSupport extends LazyLogging {
       Map.empty[AttributeName, Attribute]
   }
 
+  def replaceDataUseAttributes(existing: AttributeMap, preferred: AttributeMap): AttributeMap = {
+    // delete pre-existing DU codes, then add the DU codes from ORSP
+    (existing -
+      structuredUseRestrictionAttributeName -
+      consentCodesAttributeName --
+      allDurFieldNames.map(x => AttributeName.withLibraryNS(x))) ++ preferred
+  }
+
+  // TODO: namespaces on allDurFieldNames??!?!?
```

Comment:
Address or remove TODO?

Reply:
Ah yes, I forgot about that! This is something I noticed in pre-existing code; I will file a ticket.